joern-cli/frontends/c2cpg/src/main/scala/io/joern/c2cpg/astcreation/MacroHandler.scala

```diff
@@ -99,23 +96,66 @@ trait MacroHandler { this: AstCreator =>
     macroDef: IASTPreprocessorMacroDefinition,
     arguments: List[String]
   ): Ast = {
-    val name    = ASTStringUtil.getSimpleName(macroDef.getName)
-    val code    = node.getRawSignature.stripSuffix(";")
-    val argAsts = argumentTrees(arguments, ast).map(_.getOrElse(Ast()))
-
-    val callName     = StringUtils.normalizeSpace(name)
-    val callFullName = StringUtils.normalizeSpace(fullName(macroDef, argAsts))
-    val typeFullName = registerType(typeFor(node))
-    val callNode =
-      NewCall()
-        .name(callName)
-        .dispatchType(DispatchTypes.INLINED)
-        .methodFullName(callFullName)
-        .code(code)
-        .typeFullName(typeFullName)
-        .lineNumber(line(node))
-        .columnNumber(column(node))
-    callAst(callNode, argAsts)
+    val macroDefName     = shortName(macroDef)
+    val code             = node.getRawSignature.stripSuffix(";")
+    val typeFullName     = registerType(typeFor(node))
+    val argAsts          = argumentTrees(arguments, ast).map(_.getOrElse(Ast()))
+    val signature        = s"$typeFullName(${argAsts.size})"
+    val fileName_        = fileName(macroDef)
+    val macroDefFullName = s"$fileName_:$macroDefName:$signature"
+
+    val callNode_ =
+      callNode(node, code, macroDefName, macroDefFullName, DispatchTypes.INLINED, Some(signature), Some(typeFullName))
+
+    // we stub the corresponding method manually because we want different line/column information
+    // (call information from the actual AST node but the method gets the IASTPreprocessorMacroDefinition information)
+    createMacroMethodStub(macroDefName, macroDefFullName, signature, fileName_, argAsts.size, macroDef)
```

Comment:
This seems to be happening at the use-site of a macro. I don't think we want to create a new method stub every time a macro is used.